src/components/Inbox/index.tsx

```diff
@@ -54,6 +58,32 @@ export const InboxComponent: React.FunctionComponent<InboxComponentProps> = ({
     }
   };
 
+  const itemToggled = (item: InboxItem | undefined) => {
+    dispatch(selectInboxItem(item));
+  };
+
+  const checked = (e: any) => {
+    e.stopPropagation();
+  };
+
+  const inboxItemComponent = (item: InboxItem) => {
+    const selected = item === selectedItem;
+    const select = () => itemToggled(item);
+    const deselect = () => itemToggled(undefined);
+
+    return selected ? (
+      <InboxItemContainer button disableRipple selected onClick={deselect}>
+        <Checkbox size="small" disableRipple onClick={checked} />
+        <InboxItemText primary={item.title} secondary={item.message} />
+      </InboxItemContainer>
+    ) : (
+      <InboxItemContainer button disableRipple onClick={select}>
```

Comment:
you should pass in the `index` from the `map` on line 127 (or any sort of id you can use). im pretty sure if you open up the console, you'll see warnings. basically whenever you do a map in the render, you need to give each component a `key` to sort of act as its ID. so you would do `<InboxItemContainer ... key={index} />` here

Reply:
can you check if i did this right / did what you mean